File: src/algorithms/first-unique-char.js
/**
 * Given a string s, finds the first non-repeating character in it and returns
 * its index. If it does not exist, returns -1 instead.
 * 
 * @param {string} s - A string of lowercase English letters
 * @returns {number} - Returns the index of the first non-repeating character
 */
function firstUniqChar (s) {
  let res = -1;
  let chars = {};
  for (let i = 0; i < s.length; i++) {
    if (chars.hasOwnProperty(s[i])) {
      chars[s[i]].push(i);
    } else {
      chars[s[i]] = [i];
    }
  }
  for (let i = 0; i < s.length; i++) {
    if (chars[s[i]].length === 1) {
      res = i;
      break;
    }
  }
  return res;
}

File: src/algorithms/first-unique-char.py
from collections import Counter


def firstUniqChar(s):
    """Given a string s, find the first non-repeating character in it and 
    return its index. If it does not exist, return -1.

    Args:
        s (str): A string of lowercase English letters.

    Returns:
        int: Returns an index.
    """
    count = Counter(s)
    for key, val in count.items():
        if val == 1:
            return s.index(key)
    return -1
